### PR TITLE
tarmac: init at 0.7.0

### PR DESCRIPTION
**Pull request by wackbyte**
###### Description of changes

[Tarmac](https://github.com/Roblox/tarmac) is an asset manager for Roblox.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -24327,6 +24327,8 @@ with pkgs;
     withoutInitTools = true;
   };
 
+  tarmac = callPackage ../development/tools/tarmac { };
```
Comment (viraptor) [inline pkgs/top-level/all-packages.nix:24330]: This is the "OS-SPECIFIC" section. I don't think that was intended.

Comment (wackbyte) [reply pkgs/top-level/all-packages.nix:24330]: Where should I put it?

Comment (viraptor) [reply pkgs/top-level/all-packages.nix:24330]: Is go with the section with header `### DEVELOPMENT / TOOLS`

File: pkgs/development/tools/tarmac/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  cargoSha256 = "sha256-QnpowYv/TBXjPHK8z6KAzN3gSsfNOf9POybqsyugeWc=";
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  buildInputs = [
+    openssl
+  ];
```
Comment (viraptor) [inline pkgs/development/tools/tarmac/default.nix:28]:
```suggestion
  ] ++ lib.optionals stdenv.isDarwin [
    Security
  ];
```

File: pkgs/development/tools/tarmac/default.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, rustPlatform
+, pkg-config
+, openssl
+}:
```
Comment (viraptor) [inline pkgs/development/tools/tarmac/default.nix:7]:
```suggestion
, Security
}:
```

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -16668,6 +16668,8 @@ with pkgs;
 
   taoup = callPackage ../tools/misc/taoup { };
 
+  tarmac = callPackage ../development/tools/tarmac { };
```
Comment (viraptor) [inline pkgs/top-level/all-packages.nix:16671]:
```suggestion
  tarmac = callPackage ../development/tools/tarmac {
    inherit (darwin.apple_sdk.frameworks) Security;
  };
```

**Review by viraptor — COMMENTED** [review]
Needs some fixes to work on darwin.

**Review by viraptor — APPROVED** [review]

**Comment by viraptor** [comment]
Works. Please squash it into a single commit though. 👍 otherwise

**Comment by wackbyte** [comment]
Rebasing and pinning to `openssl_1_1` to fix the build.

**Comment by wackbyte** [comment]
Seems like I didn't do that right.

**Review by viraptor — APPROVED** [review]

**Pull request merged**